arraylist.c

```diff
@@ -87,6 +87,23 @@ array_list_add(struct array_list *arr, void *data)
 	return array_list_put_idx(arr, arr->length, data);
 }
 
+/*
+ * Deleting the idx-th element in the array_list.
+ */
+void
+array_list_del_idx(struct array_list *arr, int idx)
```

Comment:
not a hard requirement, but we nowadays prefer ```const``` modifier where possible. Helps a lot with static analyzers and protects against accidental var modification. I know it's kind of controversial with C programmers, thus not a hard requirement ;-)

Reply:
where would `const` be used ?

Reply:
```array_list_del_idx(struct array_list *const arr, const int idx)```

Reply:
Thank you, @rgerhards !  I've added "const" as you suggested.